--- qt-x11-opensource-src-4.2.2/qmake/project.cpp.QMAKEPATH	2006-11-27 11:26:03.000000000 -0600
+++ qt-x11-opensource-src-4.2.2/qmake/project.cpp	2007-02-26 13:45:39.000000000 -0600
@@ -498,6 +498,7 @@
             ret << ((*it) + concat);
     }
     ret << QLibraryInfo::location(QLibraryInfo::DataPath) + concat;
+    ret << QLibraryInfo::location(QLibraryInfo::PrefixPath) + concat;
 
     return ret;
 }
